src/app/home/home.component.ts

```diff
@@ -68,9 +68,17 @@ export class HomeComponent implements AfterViewInit, OnDestroy {
     ) {
         this.initializeURLs();
         // should any other call be made here? probably not... actually even this should be removed soon
+        let response = this.http.get(this.pingURL).subscribe(function(){
+            if(!response.ok){
+                document.getElementById('editor').disabled = true;
+                document.getElementById('editor-placeholder').innerText = 'The editor is out of service for now '
+                
+            }
+        
+        })
```

Comment:
Could you instead try 
```
this.http.get(this.pingURL).subscribe(() => {}, () => {
    // TODO add logic here
});
```

Basically this subscribe method (currently referring to a deprecated version which should be enough for now) can take three functions as parameters, with the first being called on its next value, the second is called on an error (of interest to us), and the third on completion. This link might provide some more insight on this, https://rxjs.dev/deprecations/subscribe-arguments

Reply:
I.e. the first method wouldn't do anything, and in the second method you can try adding the disabling of the editor and the text change of the placeholder.
We can refactor this later.